src/main/java/by/exadel/internship/service/impl/InterviewServiceImpl.java

```diff
@@ -99,49 +109,72 @@ public void saveInterviewForForm(UUID formId, InterviewSaveDTO interviewDTO) {
         }
     }
 
-    private void deleteTime(LocalDateTime localDateTime, UUID userId) {
-        UserDTO userDTO = userService.getById(userId);
-        List<UserTimeSlotDTO> userTimeSlotDTOList = userDTO.getUserTimeSlots()
+    private void saveForm(FormFullDTO formFullDTO){
+        log.info("Try to update Form with uuid = {}", formFullDTO.getId());
+        Form form = formRepository.findByIdAndDeletedFalse(formFullDTO.getId())
+                .orElseThrow(() -> new NotFoundException("Form with uuid = " + formFullDTO.getId() +
+                        " Not Found in DB", "form.uuid.invalid"));
+        form.setFormStatus(formFullDTO.getFormStatus());
+        Interview interview = mapper.toInterview(formFullDTO.getInterview());
+        form.setInterview(interview);
+        formRepository.save(form);
+        log.info("Successfully saved Form with uuid = {} and Interview with uuid = {}",
+                form.getId(),interview.getId());
+    }
+
+    private FormFullDTO getFormById(UUID formId){
+        Form form = formRepository.findByIdAndDeletedFalse(formId).
+                orElseThrow(() -> new NotFoundException("Form with uuid = " + formId +
+                        " Not Found in DB", "form.uuid.invalid"));
+        log.info("Return formFullDTO with uuid = {}", formId);
+        return formMapper.toFormDto(form);
+    }
+
+    private void deleteTime(LocalDateTime localDateTime, UserFullDTO userDTO) {
+        userDTO.getUserTimeSlots()
                 .stream()
                 .filter(userTimeSlotDTO -> userTimeSlotDTO.getStartDate()
-                        .equals(localDateTime)).collect(Collectors.toList());
-        userTimeSlotService.deletedById(userTimeSlotDTOList.get(0).getId());
+                        .equals(localDateTime))
+                .forEach(userTimeSlotDTO -> {
+                    userTimeSlotService.deletedById(userTimeSlotDTO.getId());
+                    userDTO.getUserTimeSlots().remove(userTimeSlotDTO);
```

Comment:
Did you check that it worked? you are removing element from collection under which forEach is executed. Not sure it will work

Reply:
Yee,  we may have exception but not all the time. I changed this and now it's working all the time

Reply:
I understand why i had exception not all the time. That was my mistake.